File: serviceos-backend/src/main/java/com/serviceos/task/api/TaskDetail.java
package com.serviceos.task.api;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record TaskDetail(TaskDirectoryItem task,UUID workflowInstanceId,UUID stageInstanceId,
        UUID workflowNodeInstanceId,String workflowNodeId,UUID workflowDefinitionVersionId,
        String workflowDefinitionDigest,UUID configurationBundleId,String configurationBundleDigest,
        String formRef,String responsibleUserId,List<String> candidateUserIds,Instant claimedAt,
        Instant startedAt,Instant completedAt,String resultRef,String resultDigest,
        List<InputVersionRef> inputVersionRefs,Instant asOf) {
    public TaskDetail { candidateUserIds=List.copyOf(candidateUserIds);inputVersionRefs=List.copyOf(inputVersionRefs); }
}

File: serviceos-admin-web/src/api/taskDetail.ts
import { apiGet } from './client'
import type { TaskDirectoryItem } from './tasksDirectory'

export type TaskDetail = {
  task: TaskDirectoryItem
  workflowInstanceId: string | null
  stageInstanceId: string | null
  formRef: string | null
  responsibleUserId: string | null
  candidateUserIds: string[]
  claimedAt: string | null
  startedAt: string | null
  completedAt: string | null
  resultRef: string | null
  resultDigest: string | null
  asOf: string
}

export function getAuthorizedTask(taskId: string) {
  return apiGet<TaskDetail>(`/tasks/${taskId}`)
}
